### PR TITLE
Various fixes on absl::Status migration

### PR DESCRIPTION
**Pull request by veblush**
This is a collection of fixes addressing the issues found when enabling `GRPC_ERROR_IS_ABSEIL_STATUS`. Most of them are to fill the gap between grpc_error and absl::Status.

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/27557)

### CODE REVIEW

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 1]
```diff
@@ -596,6 +596,7 @@ grpc_error_handle Chttp2ServerListener::Create(
   // The bulk of this method is inside of a lambda to make cleanup
   // easier without using goto.
   grpc_error_handle error = [&]() {
+    grpc_error_handle error = GRPC_ERROR_NONE;
```
Comment (yashykt) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:599]: how was this working earlier?

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 2]
```diff
@@ -861,6 +862,7 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
   std::vector<grpc_error_handle> error_list;
   // Using lambda to avoid use of goto.
   grpc_error_handle error = [&]() {
+    grpc_error_handle error = GRPC_ERROR_NONE;
```
Comment (yashykt) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:865]: same question? why is this needed? how was it working earlier?

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 1 above)
Comment (veblush) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc:599]: Previously outer `error` was captured and used so it worked in a strange way but it somehow worked. I guess It's mainly acceptable because it's a c pointer. With a C++ instance, however, it becomes more strict and it stops working in this way.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 2 above)
Comment (veblush) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc:865]: Ditto with above.
